android/modules/geolocation/src/java/ti/modules/titanium/geolocation/TiLocation.java

```diff
@@ -106,8 +109,9 @@ public boolean getLocationServicesEnabled()
 			}
 		}
 
+		//TIMOB-23135	Android: Ti.Geolocation.locationServicesEnabled returns false, but works
 		for (String providerName : providerNames) {
-			if (providerName.equals(LocationManager.GPS_PROVIDER)) {
+			if (providerName.equals(LocationManager.GPS_PROVIDER) || providerName.equals(LocationManager.NETWORK_PROVIDER)) {
```

Comment:
Question:
Is there a reason why we're limiting this to GPS and NETWORK? Does disabling location services on a phone not disable the passive provider?

I ask because our "GeolocationModule" reads from the PASSIVE location provider.
https://github.com/appcelerator/titanium_mobile/blob/master/android/modules/geolocation/src/java/ti/modules/titanium/geolocation/GeolocationModule.java#L200

Reply:
Okay, so @maggieaxway and I just talked about this. The PASSIVE provider will still provide data, even if location services is disabled. In fact, that was a customer complaint here...
https://jira.appcelerator.org/browse/TIMOB-23135

So, the real question here is **should** we provide PASSIVE location data at all? The GPS and NETWORK providers are the ones we really want. I'm starting to think reverting the code like Maggie just did now and removing PASSIVE provider support might be the way to go.

@garymathews, you've been playing around with this recently, what are your thoughts? How is this compared to Google Play Service's fused location provider?